transport.go

```diff
@@ -45,14 +45,22 @@ func Chain(base http.RoundTripper, mw ...func(http.RoundTripper) http.RoundTripp
 		base = http.DefaultTransport
 	}
 
+	// Filter out nil transports.
+	filtered := []func(http.RoundTripper) http.RoundTripper{}
+	for _, fn := range mw {
+		if fn != nil {
+			filtered = append(filtered, fn)
+		}
+	}
```

Comment:
We can do it like this, this will remove `nil` elements. But only for version 1.21 an above, transport is 1.14.
```suggestion
	mw = slices.DeleteFunc(mw, func(n func(http.RoundTripper) http.RoundTripper) bool {
		return n == nil
	})
```

If you don't like it, then at least rename `filtered` to `nonNilMiddlewares`, `filtered` doesn't seems right name to me.

Reply:
We can't use `slices.DeleteFunc` without breaking compatibility with older Go versions.

We're now compatible with Go 1.14, same as github.com/go-chi/chi.